### DISCUSSION TITLE
Top-level country search

### DISCUSSION THREAD

**Discussion by mausch** [osm-search/Nominatim#2675] [Q&A]
Hi, what data do I need to import into my Nominatim instance to resolve country names?
e.g. in the public instance I can query this and get the expected result: https://nominatim.openstreetmap.org/ui/search.html?q=united+states
But if I run that query on my instance I get some result in South Africa just because I have imported https://download.geofabrik.de/africa/south-africa.html but I don't have the US in my database.
I don't want to import the whole planet but surely it should be possible to resolve country names without needing to import everything?

Is this what the [Fallback Country Boundaries](https://nominatim.org/release-docs/3.4.0/data-sources/Country-Grid/#fallback-country-boundaries) is for? I'm on Nominatim 4 but the [equivalent docs](https://nominatim.org/release-docs/latest/develop/data-sources/#country-grid) aren't very clear about it.

Otherwise is there a dataset available for this?

Many thanks

**Comment by lonvia** [comment]
Nominatim always needs an OSM object to return. If you only import an extract, then it can still use the fallback names, to find out that you where looking for the country, but it doesn't have the relevant data to return. I admit that this is not very satisfying.

Nominatim could return an artificial object instead made up of the information from the country_name table. That would need to be implemented, of course.

Another quick-fix solution for you is to grab the country relations from OSM and import those into your existing database using `nominatim add-data --file`.

**Reply by mausch** [reply]
Thanks, that pretty much confirms what I was thinking! I was hoping I was missing something obvious here :slightly_smiling_face: 

> Another quick-fix solution for you is to grab the country relations from OSM and import those into your existing database using nominatim add-data --file.

Yep, I've been looking for this and found https://osm-boundaries.com/Map ... would this be useful here? I have to click on every single country which somehow tells me it's not the right tool for this?
Also it downloads a geojson file... I'm sure I can figure out how to convert this to something that can be imported into Nominatim, but is this really the way to go?
Or maybe I should be getting this data from somewhere else?

In other words, this seems like a fairly common problem, just wondering if there's a well-known solution to this.

Many thanks

**Reply by mtmail** [reply]
I'm not aware of a country-level-only extract. You probably need the full planet file from https://planet.openstreetmap.org/ and filter it, for example with https://osmcode.org/osmium-tool/manual.html#filtering-by-tags

```
time osmium tags-filter planet-220411.osm.pbf r/admin_level=2 -o admin_level_2-planet-220411.osm.pbf
[======================================================================] 100%
real	6m45.980s

ls -lah *.pbf
  64G planet-220411.osm.pbf
  32M admin_level_2-planet-220411.osm.pbf
```

It might contain too many places, sometimes embassies are marked as `admin_level=2` (https://wiki.openstreetmap.org/wiki/Tag:admin_level%3D2). So further filtering by `boundary=administrative` or `ISO3166-1=*` could work.

**Reply by lonvia** [reply]
Overpass is another option. Download all relations with boundary=administrative and admin_level=2. Something along those lines: https://overpass-api.de/api/interpreter?data=relation%5B%22boundary%22%3D%22administrative%22%5D%5B%22admin%5Flevel%22%3D%222%22%5D%3B%28%2E%5F%3B%3E%3B%29%3Bout%3B%0A

You need OSM XML or PBF to be able to import it into Nominatim.
